util/logs/workflow-logger.go

```diff
@@ -80,6 +81,14 @@ func WorkflowLogs(ctx context.Context, wfClient versioned.Interface, kubeClient
 
 	var podListOptions metav1.ListOptions
 
+	// get env variable for pod logs redaction
+	enablePodLogRedaction := os.Getenv("ARGO_REDACT_POD_LOGS")
+	// get secrets for redaction
+	secrets, err := kubeClient.CoreV1().Secrets(req.GetNamespace()).List(ctx, metav1.ListOptions{})
```

Comment:
I don't think we can list all the secrets. See https://github.com/argoproj/argo-workflows/issues/8534

Reply:
@terrytangyuan I can see the [GetSecret](https://github.com/argoproj/argo-workflows/blob/master/server/cache/resource_cache.go#L42) method is available but it requires the service account name and we can only fetch the secrets associated with a service account using this method.

One approach is using [Service account Lister](https://github.com/argoproj/argo-workflows/blob/master/server/cache/resource_cache.go#L27) to get all the service accounts and fetch the secrets of each SA using above mentioned GetSecret method. Although we might not have all the secrets with this approach.

Can you suggest any other alternate approaches?

Reply:
@terrytangyuan @anilkumar-pcs 
I'd like to resolve #8685 but I don't have any better idea then 
1. list secrets and keep then in cache or queue
2. when command like "echo" or "export", check command if they include one of those secrets
3. hide secrets

but I think those feature make large burden on argo-workflows.... 
any suggestion on this??